### ISSUE TITLE
/tmp/v is restricted to root

### ISSUE THREAD

**Issue by steve-kaufman** [vlang/v#8684]
**V version:** 
0.2.2 (09cff69)

**OS:** 
Debian (WSL)

**What did you do?**
I installed V by downloading v_linux.zip from vlang.io, unzipping the file into `$HOME/v`, and running `sudo ./v symlink` in that directory.

**What did you expect to see?**
I expected that v would work properly using non-root permissions.
 
**What did you see instead?**
When attempting to use `v run`, I encountered the error:
```
V panic: failed to open file "/tmp/v/main.6055694134605297992.tmp.c"
```
Upon further examination, it looks like not only is `/tmp/v` (and everything in it) owned by root, but its permissions are read-write for root and read only for all others.

**Workaround**
Force deleting `/tmp/v` (i.e. `sudo rm -rf /tmp/v`) and rerunning a `v` command seems to fix the issue, presumably because it recreates `/tmp/v` and performs the necessary operations with user permissions instead of root permissions. 

**Possible Fix**
The `v symlink` command appears to be the culprit, as it's creating the `/tmp/v` directory as root. Perhaps that creation isn't necessary at all since `v` seems to create it automatically if it isn't there?

You could also consider a more conventional install process that doesn't involve binaries being stored in user space.

**Comment by clubby789** [comment]
Seems like implementing `umask` into the `os` library would be helpful for this.

**Comment by JalonSolov** [comment]
More like the `symlink` command needs to do a `chown` if it created the directory.  It isn't actually `symlink`'s fault... it's because you have to run it with `sudo`, and that make it run _as_ root.

**Comment by JalonSolov** [comment]
Actually, even simpler would be for `symlink` to do a recursive delete of `/tmp/v` when it finishes.  It's just a temporary directory, so that wouldn't hurt anything.

**Comment by spytheman** [comment]
The temporary folder removal after `v symlink` is implemented in 252074836.
